proto/compiler.bzl

```diff
@@ -32,10 +32,16 @@ load(
 
 GoProtoCompiler = provider()
 
-def go_proto_compile(go, compiler, proto, imports, importpath):
+def go_proto_compile(go, compiler, protos, imports, importpath):
     go_srcs = []
     outpath = None
-    for src in proto.check_deps_sources:
+    dep_sources = depset(direct = [], transitive = [proto.check_deps_sources for proto in protos])
+    transitive_descriptor_sets = depset(
+        direct = [],
+        transitive = [proto.transitive_descriptor_sets for proto in protos],
+    )
+
+    for src in dep_sources:
```

Comment:
`dep_sources.to_list()`

Bazel will eventually make it an error to iterate a `depset`. There is a flag `--incompatible_depset_is_not_iterable`, which detects this. (My previous code here is wrong; I didn't realize `proto.check_deps_sources` was a `depset`).